src/lib.rs

```diff
@@ -38,354 +43,137 @@ pub struct LongLivedToken {
 impl HomeAssistantAPI {
     pub fn new(instance_urls: Vec<String>) -> Self {
         return Self {
-            instance_urls: instance_urls,
+            instance_urls,
             client: reqwest::Client::new(),
             token: None,
             webhook_id: None,
             cloudhook_url: None,
             remote_ui_url: None,
-        }
+        };
     }
 
-    pub fn auth_token(
-        &mut self,
-        oauth_token: String,
-        refresh_token: String,
-        token_expiration: u64,
-    ) -> Result<(), errors::Error> {
-        let oauth = OAuthToken {
-            token: oauth_token,
-            refresh_token: refresh_token,
-            token_expiration: (time::UNIX_EPOCH + time::Duration::from_secs(token_expiration)),
+    pub fn single_instance(instance_url: String) -> Self {
+        return Self {
+            instance_urls: vec![instance_url],
```

Comment:
What's the thought behind multiple instance URLs? For local/remote? Looks like most of the below code just does `[0]`

Reply:
This is more based off of @BCNelson work where he had multiple instances. I'm open to just using one.